Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -542,6 +542,31 @@ void HitTestResult::enterFullscreenForVideo() const
 #endif
 }
 
+bool HitTestResult::mediaIsInVideoViewer() const
+{
+#if PLATFORM(MAC) && ENABLE(VIDEO) && ENABLE(VIDEO_PRESENTATION_MODE)
+    HTMLMediaElement* mediaElt(mediaElement());
+    return is<HTMLVideoElement>(mediaElt) && mediaElt->fullscreenMode() == HTMLMediaElementEnums::VideoFullscreenModeInWindow;
+#else
+    return false;
+#endif
+}
+
+void HitTestResult::toggleVideoViewer() const
+{
+#if PLATFORM(MAC) && ENABLE(VIDEO) && ENABLE(VIDEO_PRESENTATION_MODE)
+    RefPtr mediaElement(this->mediaElement());
+    RefPtr videoElement = dynamicDowncast<HTMLVideoElement>(*mediaElement);
```

Comment:
I think this is unsafe is `mediaElement` is ever null?

Reply:
I think dynamicDowncast returns a nullptr if mediaElement is null

Reply:
I think you need to do:
```
RefPtr videoElement = dynamicDowncast<HTMLVideoElement>(mediaElement);
```
I.e., without the `*` dereference.